### ISSUE TITLE
Allele molecular details display error for an insertion/deletion complex substitution allele tm1133

### ISSUE THREAD

**Issue by wormbasehelp** [WormBase/website#6851]
*Help Desk chat transcript. Issue can be closed if question was resolved in chat.*

Chris: Hello Amy, how can I help?</br>visitor_name (visitor_email): Hi, i think there might be an error with the mutation sequence for the tm1133 variation</br>Chris: Oh, OK, thanks for letting us know. What seems to be the problem?</br>visitor_name (visitor_email): it says that it should be a 419bp deletion and a 14bp insertion but the mutation sequence doesn't show the deletion</br>visitor_name (visitor_email): it looks like a substitution of the same length instead</br>visitor_name (visitor_email): i remember looking up this same variation a couple of weeks ago before the format was changed and it did show a deletion</br>Chris: OK, I'll forward your message to our allele/variation curator who can get back to you. We did just update our display of allele molecular details, so there may be a bug in the display code. Thanks for letting us know. We'll get back to you soon.\</br>Chris: Is there anything else I can help you with?</br>visitor_name (visitor_email): thank you so much please let me know as soon as you can</br>Chris: OK, yes I will</br>visitor_name (visitor_email): that's all thank you :)</br>Chris: OK, thanks again. Have a great day!\</br>visitor_name (visitor_email): you too</br>Chris: Thanks!

**Reported by:** Amy **** (amyy***************)
**Submitted from:** <a target="_blank" href="http://www.wormbase.org/https://www.wormbase.org/species/c_elegans/variation/WBVar00250150#0-46572-5">https://www.wormbase.org/species/c_elegans/variation/WBVar00250150#0-46572-5</a>
**Browser:** Firefox 65.0

**Comment by chris-grove** [comment]
@a8wright @sibyl229 
Could you have a look at this allele page with the new Molecular Details display?

https://www.wormbase.org/species/c_elegans/variation/WBVar00250150#05--10

This allele is supposed to be a 419 bp deletion with a 14bp insertion. It looks like the wildtype sequence is displayed correctly (although it does have a hyphen "-" just after the first flanking sequence). The mutant sequence appears to have the 14bp insertion at the beginning of the mutated region, but then includes the entire 419 bp sequence that should be omitted in the mutant.

**Comment by a8wright** [comment]
I will try to look into this and get it fixed this morning.

**Comment by Paul-Davis** [comment]
The live site display looks wrong where the staging site looks like what we would expect (without lookin gin great detail).

Live: https://www.wormbase.org/species/c_elegans/variation/WBVar00250150#02-4581-10

![live](https://user-images.githubusercontent.com/717707/51262290-fc9fa880-19a9-11e9-8150-434c0935f3c5.png)

https://staging.wormbase.org/species/c_elegans/variation/WBVar00250150#02-4581-10

![staging](https://user-images.githubusercontent.com/717707/51262309-06291080-19aa-11e9-9d1b-801ec3032098.png)

Data in the database:
Flanking_sequences: tctgaattttcgtcactcagtgcgtgtcca tagttcagccacattgtctttcagttctcc
Insertion: GTTCAGCCACATTG
+ve strand annotation.

**Comment by a8wright** [comment]
@Paul-Davis thank you for looking on stage. I checked what the acedb code was creating and replicated it. You are correct staging is now fixed. Fro this exact variation the cache has been deleted in production. So it is fixed there aswell. For other variants that are affected, we will see the fix within a day. 

@chris-grove this was caused by me making the assumption that substitutions handled all cases where there were insertions and deletions. So the logic in the code was wrong because of a bad assumption not really a bug. I am glad I listened in to the call where we were discussing "delins". If we had this call a couple weeks ago I wouldn't have made the assumption!
